### PR TITLE
Kiarash2077 patch 8

### LINKED ISSUE #17: [Part 2] Suggest Tasks
Suggest Tasks: Suggests tasks based on priority and approaching deadlines.

example

```
Advanced To-Do List Application 
1. Add Task 
2. Remove Task 
3. View Tasks 
4. Suggest Tasks 
5. Exit 
Enter your choice: 4 
Good afternoon! Here are some tasks you might want to work o n: 
Buy groceries - high - 2024-06-10 
Finish homework - medium - 2024-06-15 
```

### PR DESCRIPTION
**Pull request by kiarash2077**
close #17

### CODE REVIEW

File: to_do_list_part_two.py [hunk 1]
```diff
@@ -27,22 +27,23 @@ def view_tasks(task_list):
         print("The to-do list is empty. Consider adding a new task!")
     else:
         print("To-Do List:")
-        priority_order = {'high': 1, 'medium': 2, 'low': 3}
-        sorted_tasks = sorted(task_list, key=lambda x: (priority_order[x['priority']], x['deadline']))
+        sorted_tasks = sort_tasks_by_priority_and_deadline(task_list)
         for idx, task in enumerate(sorted_tasks, start=1):
-            print(f"{idx}. {task['task']} - {task['priority']} - {task['deadline'].strftime('%Y-%m-%d')}")
+            print(f"{idx}. {task['task']} - {task['priority']} - {task['deadline']}")
```
Comment (shigeokitamura) [inline to_do_list_part_two.py]: https://github.com/kiarash2077/To-Do-List-Application/pull/21#discussion_r1803850313

This change is not reflected in this pull request.

```suggestion
            print(f"{idx}. {task['name']} - {task['priority']} - {task['deadline']}")
```

File: to_do_list_part_two.py [hunk 2]
```diff
@@ -27,22 +27,23 @@ def view_tasks(task_list):
...
     else:
-        print("Greetings! Here are some taskes you might want to work on:")
-        priority_order = {'high': 1, 'medium'': 2, "low': 3}
-        sorted_tasks = sorted(task_list, key=lambda x: (priority_order[x['priority']], x['deadline']))
+        print("Good afternoon! Here are some tasks you might want to work on:")
+        sorted_tasks = sort_tasks_by_priority_and_deadline(task_list)
         for task in sorted_tasks:
-            print(f"{task['task']} - {task['priority']} - {task['deadline'].strftime('%Y-%m-%d')}")
-
+            print(f"{task['task']} - {task['priority']} - {task['deadline']}")
```
Comment (shigeokitamura) [inline to_do_list_part_two.py]:
```suggestion
            print(f"{task['name']} - {task['priority']} - {task['deadline']}")
```

File: to_do_list_part_two.py [hunk 3]
```diff
@@ -27,22 +27,23 @@ def view_tasks(task_list):
...
+
```
Comment (shigeokitamura) [inline to_do_list_part_two.py]: Please remove unnecessary spaces

**Review by shigeokitamura — CHANGES_REQUESTED** [review]
Can you add `close #17` to the description?

File: to_do_list_part_two.py [hunk 4]
```diff
@@ -27,22 +27,23 @@ def view_tasks(task_list):
...
-        priority_order = {'high': 1, 'medium': 2, 'low': 3}
-        sorted_tasks = sorted(task_list, key=lambda x: (priority_order[x['priority']], x['deadline']))
+        sorted_tasks = sort_tasks_by_priority_and_deadline(task_list)
         for idx, task in enumerate(sorted_tasks, start=1):
-            print(f"{idx}. {task['task']} - {task['priority']} - {task['deadline'].strftime('%Y-%m-%d')}")
+            print(f"{idx}. {task['task']} - {task['priority']} - {task['deadline']}")
 
+def sort_tasks_by_priority_and_deadline(task_list):
+    priority_order = {'high': 1, 'medium': 2, 'low': 3}
+    return sorted(task_list, key=lambda x: (priority_order.get(x['priority'], 4), x['deadline']))
```
Comment (shigeokitamura) [inline to_do_list_part_two.py:35]: Sorting looks good

```
    task_list = [
        {"name": "task_1", "priority": "high", "deadline": "2024-10-01"},
        {"name": "task_2", "priority": "medium", "deadline": "2024-10-01"},
        {"name": "task_3", "priority": "medium", "deadline": "2024-10-02"},
        {"name": "task_4", "priority": "high", "deadline": "2024-10-02"},
        {"name": "task_5", "priority": "high", "deadline": "2024-10-03"},
        {"name": "task_6", "priority": "medium", "deadline": "2024-10-03"}
    ]
```

```
To-Do List:
1. task_1 - high - 2024-10-01
2. task_4 - high - 2024-10-02
3. task_5 - high - 2024-10-03
4. task_2 - medium - 2024-10-01
5. task_3 - medium - 2024-10-02
6. task_6 - medium - 2024-10-03
```

File: to_do_list_part_two.py (hunk 1 above)
Comment (kiarash2077) [reply to_do_list_part_two.py]: you mean it is not working?

File: to_do_list_part_two.py [hunk 5]
```diff
@@ -27,22 +27,21 @@ def view_tasks(task_list):
         print("The to-do list is empty. Consider adding a new task!")
     else:
         print("To-Do List:")
-        priority_order = {'high': 1, 'medium': 2, 'low': 3}
-        sorted_tasks = sorted(task_list, key=lambda x: (priority_order[x['priority']], x['deadline']))
+        sorted_tasks = sort_tasks_by_priority_and_deadline(task_list)
         for idx, task in enumerate(sorted_tasks, start=1):
-            print(f"{idx}. {task['task']} - {task['priority']} - {task['deadline'].strftime('%Y-%m-%d')}")
-
+            print(f"{idx}. {task['name']} - {task['priority']} - {task['deadline']}")
```
Comment (shigeokitamura) [inline to_do_list_part_two.py]: There is an unnecessary whitespace in the last of this line that should be deleted.

File: to_do_list_part_two.py [hunk 6]
```diff
@@ -27,22 +27,21 @@ def view_tasks(task_list):
...
     else:
-        print("Greetings! Here are some taskes you might want to work on:")
-        priority_order = {'high': 1, 'medium'': 2, "low': 3}
-        sorted_tasks = sorted(task_list, key=lambda x: (priority_order[x['priority']], x['deadline']))
+        print("Good afternoon! Here are some tasks you might want to work on:")
+        sorted_tasks = sort_tasks_by_priority_and_deadline(task_list)
         for task in sorted_tasks:
-            print(f"{task['task']} - {task['priority']} - {task['deadline'].strftime('%Y-%m-%d')}")
-
+            print(f"{task['name']} - {task['priority']} - {task['deadline']}")
```
Comment (shigeokitamura) [inline to_do_list_part_two.py]: There is an unnecessary whitespace in the last of this line that should be deleted.

Comment (shigeokitamura) [inline to_do_list_part_two.py]: Can you add a blank line between the functions?

**Review by shigeokitamura — CHANGES_REQUESTED** [review]

**Review by shigeokitamura — APPROVED** [review]
